lib/matplotlib/backends/backend_gtk3.py

```diff
@@ -501,11 +501,14 @@ def _init_toolbar(self):
             image = Gtk.Image()
             image.set_from_file(
                 str(cbook._get_data_path('images', image_file + '.png')))
-            self._gtk_ids[text] = tbutton = Gtk.ToolButton()
+            self._gtk_ids[text] = tbutton = (
+                Gtk.ToggleToolButton() if callback in ['zoom', 'pan'] else
+                Gtk.ToolButton())
             tbutton.set_label(text)
             tbutton.set_icon_widget(image)
             self.insert(tbutton, -1)
-            tbutton.connect('clicked', getattr(self, callback))
+            tbutton._signal_handler = tbutton.connect(
```

Comment:
is `_signal_handler` an API of GTK or is it just some attibute that you invented to carry the signal handler around. If the latter, that should be documented in a commment.

Reply:
it's just our own thing; comment added.